daal4py/sklearn/_utils.py

```diff
@@ -151,3 +152,113 @@ def get_number_of_types(dataframe):
         return len(set(dtypes))
     except TypeError:
         return 1
+
+
+def _create_global_queue_from_sycl_context():
+    import sys
+    d4p_target = None
+    if 'daal4py.oneapi' in sys.modules:
+        from daal4py.oneapi import _get_device_name_sycl_ctxt
+        d4p_target = _get_device_name_sycl_ctxt()
+
+    if d4p_target is not None:
+        try:
+            from dpctl import SyclQueue
+        except ImportError:
+            raise RuntimeError("dpctl need to be installed for device offload")
+        return SyclQueue(d4p_target if d4p_target != 'host' else 'cpu')
+    return None
+
+
+def _transfer_to_host(queue, *data):
+    has_usm_data = False
+
+    host_data = []
+    for item in data:
+        usm_iface = getattr(item, '__sycl_usm_array_interface__', None)
+        if usm_iface is not None:
+            import dpctl.memory as dp_mem
```

Comment:
Check available dpctl?

Reply:
There is no way to process this case without dpctl. So this is requirement to have it

Reply:
For  user application? Then we always carry dependency DPCPP, and again, for all applications - include only CPU applications, our library will have size more than 1GB. There are other options?